org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/capabilities/XMLCapabilityManager.java

```diff
@@ -0,0 +1,35 @@
+package org.eclipse.lsp4xml.settings.capabilities;
+
+import org.eclipse.lsp4j.ClientCapabilities;
+
+/**
+ * ClientPreferences
+ */
+public class XMLCapabilityManager {
+
+	ClientCapabilityPreferences clientCapabilityPreferences;
+	ClientCapabilitiesWrapper clientWrapper;
+
+	public void setClientCapabilityPreferences(ClientCapabilityPreferences clientCapabilityPreferences) {
+		this.clientCapabilityPreferences = clientCapabilityPreferences;
+	}
+
+	public ClientCapabilityPreferences getClientCapabilityPreferences() {
+		return this.clientCapabilityPreferences;
+	}
+
+	public void updateClientCapabilitiesWrapper(ClientCapabilities clientCapabilities) {
+		boolean dynamicRegistrationSupported = clientCapabilities.getTextDocument() != null;
```

Comment:
calling it dynamicRegistrationSupported is a bit dubious. Not sure about that one.